src/devices/LedDisplay/Large4Digit7SegmentDisplay.cs

```diff
@@ -0,0 +1,310 @@
+    /// </remarks>
+    public class Large4Digit7SegmentDisplay : Ht16k33, ISevenSegmentDisplay
+    {
+        #region Private members
+        #region Constants
+        /// <summary>
+        /// Number of digits supported by display
+        /// </summary>
+        private const int MaxNumberOfDigits = 4;
+
+        /// <summary>
+        /// This display does not support dot bits for each digit
+        /// </summary>
+        private const byte SegmentMask = 0b0111_1111;
+        #endregion
+
+        #region Enums
+        /// <summary>
+        /// Digit address within display buffer
+        /// </summary>
+        private enum Address
+        {
+            /// <summary>
+            /// First digit
+            /// </summary>
+            Digit1 = 1,
+            /// <summary>
+            /// Second digit
+            /// </summary>
+            Digit2 = 3,
+            /// <summary>
+            /// Dot setting bits
+            /// </summary>
+            Dots = 5,
+            /// <summary>
+            /// Third digit
+            /// </summary>
+            Digit3 = 7,
+            /// <summary>
+            /// Fourth digit
+            /// </summary>
+            Digit4 = 9
+        }
+        #endregion
+
+        /// <summary>
+        /// List of digit addresses for sequential writing
+        /// </summary>
+        private static readonly Address[] s_digitAddressList = new[] { Address.Digit1, Address.Digit2, Address.Digit3, Address.Digit4 };
+
+        /// <summary>
+        /// Empty display buffer
+        /// </summary>
+        private static readonly byte[] s_clearBuffer = new byte[]
+        {
+            0x00, 0b0000_0000,
+            0x01, 0b0000_0000,
+            0x02, 0b0000_0000,
+            0x03, 0b0000_0000,
+            0x04, 0b0000_0000
+        };
+
+        /// <summary>
+        /// Display buffer
+        /// </summary>
+        private readonly byte[] _displayBuffer = new byte[]
+        {
+            0x00, 0b0000_0000,
+            0x01, 0b0000_0000,
+            0x02, 0b0000_0000,
+            0x03, 0b0000_0000,
+            0x04, 0b0000_0000
+        };
+
+        /// <summary>
+        /// Translate digit number to buffer address
+        /// </summary>
+        /// <param name="digit">digit to translate</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="digit"/></exception>
+        private static int TranslateDigitToBufferAddress(int digit) => digit switch
+        {
+            0 => (int)Address.Digit1,
+            1 => (int)Address.Digit2,
+            2 => (int)Address.Digit3,
+            3 => (int)Address.Digit4,
+            _ => throw new ArgumentOutOfRangeException(nameof(digit)),
+        };
+        #endregion
+
+        #region Public members
+        #region Enums
+        /// <summary>
+        /// Dot display setup
+        /// </summary>
+        /// <remarks>Specific to device</remarks>
+        [Flags]
+        public enum Dot : byte
```

Comment:
this should not be nested

Reply:
This Dot configuration is specific to the backpack device. Should I move it out of the class and rename it to `Large4Digit7SegmentDisplayDot` to maintain namespacing?

Reply:
yes, let's move it out